master/internal/api_trials.go

```diff
@@ -695,58 +672,99 @@ func (a *apiServer) MultiTrialSample(trialID int32, metricNames []string,
 			var metric apiv1.SummarizedMetric
 			metric.Name = name
 			metricMeasurements, err = a.m.db.TrainingMetricsSeries(
-				trialID, startTime, name, startBatches, endBatches, xAxisLabelMetrics)
+				trialID, startTime, name, startBatches, endBatches,
+				xAxisLabelMetrics,
+				maxDatapoints)
 			if err != nil {
 				return nil, errors.Wrapf(err, "error fetching time series of training metrics")
 			}
 			metric.Type = apiv1.MetricType_METRIC_TYPE_TRAINING
 
-			metricSeriesTime = lttb.Downsample(metricMeasurements.Time, maxDatapoints, logScale)
-			a.formatMetricsTime(&metric, metricSeriesTime)
-			metricSeriesBatch = lttb.Downsample(metricMeasurements.Batches, maxDatapoints, logScale)
-			a.formatMetricsBatch(&metric, metricSeriesBatch)
+			a.formatMetrics(&metric, metricMeasurements)
 
 			// For now "epoch" is the only custom xAxis metric label supported so we
 			// build the `MetricSeriesEpoch` array. In the future this logic should
 			// be updated to support any number of xAxis metric options
-			metricSeriesEpoch = lttb.Downsample(metricMeasurements.AverageMetrics["epoch"],
-				maxDatapoints, logScale)
-			a.formatMetricsEpoch(&metric, metricSeriesEpoch)
 
-			if len(metricSeriesBatch) > 0 || len(metricSeriesTime) > 0 {
+			if len(metricMeasurements) > 0 {
 				metrics = append(metrics, &metric)
 			}
 		}
 		if (metricType == apiv1.MetricType_METRIC_TYPE_VALIDATION) ||
 			(metricType == apiv1.MetricType_METRIC_TYPE_UNSPECIFIED) {
 			var metric apiv1.SummarizedMetric
 			metric.Name = name
-			metricMeasurements, err = a.m.db.ValidationMetricsSeries(
-				trialID, startTime, name, startBatches, endBatches, xAxisLabelMetrics)
+			metricMeasurements, err = trials.ValidationMetricsSeries(
+				trialID, startTime, name, startBatches, endBatches,
+				xAxisLabelMetrics, maxDatapoints, rangeType, integerRange,
+				timeRange)
 			if err != nil {
 				return nil, errors.Wrapf(err, "error fetching time series of validation metrics")
 			}
 			metric.Type = apiv1.MetricType_METRIC_TYPE_VALIDATION
+			a.formatMetrics(&metric, metricMeasurements)
 
-			metricSeriesTime = lttb.Downsample(metricMeasurements.Time, maxDatapoints, logScale)
-			a.formatMetricsTime(&metric, metricSeriesTime)
-			metricSeriesBatch = lttb.Downsample(metricMeasurements.Batches, maxDatapoints, logScale)
-			a.formatMetricsBatch(&metric, metricSeriesBatch)
-			metricSeriesEpoch = lttb.Downsample(metricMeasurements.AverageMetrics["epoch"],
-				maxDatapoints, logScale)
-			a.formatMetricsEpoch(&metric, metricSeriesEpoch)
-
-			if len(metricSeriesBatch) > 0 || len(metricSeriesTime) > 0 {
+			if len(metricMeasurements) > 0 {
 				metrics = append(metrics, &metric)
 			}
 		}
 	}
+	if len(metricIds) > 0 {
```

Comment:
just to clarify, I meant 
- the function can have both in the parametes / protos
- internally, when both are specified, it should error out. right now if you specify both together in one call it'd combine them in a certain way - I don't think we should support that.
- since metric ids are a superset of existing metric names, the function could convert metric names into metric ids internally, and then have unified sampling code working with metric ids

Reply:
I have the first two bullet points covered now. 

The last bullet point also makes sense. The only concern there currently is making sure that everywhere on the front end where we call this API has both specified `metricType` and `metricName` and if not we need to decide how to handle that specific case, which I assume would be to just search both training and validation metrics for now if we are only given a `metricName`. 

The switch in my mind would not be too problematic but would require some investigation and possible refactoring. My initial thought is to keep things as is, and make the switch later as we move both the FE and BE to more unified metrics but I am open to discussion.

Reply:
idk if this is an endpoint that is used much (think this was created purely for trial comparison project that was tabled?) so no strong opinions, but thinking we could just use `metricIds` only going forward and drop the `metricName` and `metricType` params?

Reply:
I believe this API is in use by the main trail metrics chart, so it's used on every trial page, quite a lot.